apache/gobblin
# [GOBBLIN-1859] Multi-active Unit Test for Multi-Participant state

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!


### JIRA
- [X] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1859 


### Description
- [X] Here are some details about my PR, including screenshots (if applicable):
In multi-active mode, multiple participants will be reading/writing to MySQL table. We want to make sure each participant acts on the table in a manner that takes into account that the state may have changed while the MultiActiveLeaseArbiter is processing the result of a READ. This PR adds unit tests that validate SQL Insertion statements are conditional upon the state of the particular row corresponding to the flow action event being unchanged from the read made by this participant. 

### Tests
The PR adds 3 unit tests to test each of the cases of acquiring a lease to ensure the following insertions do not complete if the state has changed from our initial read. 
- insert when there is no matching primary key in the table
- insert if a lease has expired
- insert if a lease has completed 

### Commits
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"
